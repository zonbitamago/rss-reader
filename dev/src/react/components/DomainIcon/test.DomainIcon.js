import React from "react";
import { shallow } from "enzyme";

import DomainIcon from "./DomainIcon";

describe("DomainIcon", () => {
  let component, props;

  beforeEach(() => {
    props = { src: "http://www.google.com/s2/favicons?domain=qiita.com" };
    component = shallow(<DomainIcon {...props} />);
  });

  it("should", () => {
    expect(component).toMatchSnapshot();
  });
});
