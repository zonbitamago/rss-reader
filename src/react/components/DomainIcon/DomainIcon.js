import React, { Component } from "react";
import PropTypes from "prop-types";
import styles from "./DomainIcon.css";

class DomainIcon extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div className="DomainIcon">
        <img alt={this.props.alt} src={this.props.src} />
      </div>
    );
  }
}

DomainIcon.propTypes = {
  src: PropTypes.string.isRequired,
  alt: PropTypes.string
};

DomainIcon.defaultProps = {};

export default DomainIcon;
