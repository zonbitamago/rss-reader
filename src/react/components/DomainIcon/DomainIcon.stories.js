import React from "react";
import { storiesOf } from "@storybook/react";

import DomainIcon from "./DomainIcon";

storiesOf("Atoms/DomainIcon", module).add("DomainIcon", () => (
  <DomainIcon src="http://www.google.com/s2/favicons?domain=qiita.com" />
));
