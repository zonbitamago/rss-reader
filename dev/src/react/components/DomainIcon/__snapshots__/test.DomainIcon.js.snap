// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DomainIcon should 1`] = `
<div
  className="DomainIcon"
>
  <img
    src="http://www.google.com/s2/favicons?domain=qiita.com"
  />
</div>
`;
